### ISSUE TITLE
Renaming temporary file failed (file exists)

### ISSUE THREAD

**Issue by chris-y** [salass00/zip_lib#1]
zip.library v53.11

When adding or deleting entries from Zip files, I get the above error.

My code is as follows:
```
	int err = 0;
	zip_t *zip = zip_open(archive, 0, &err);

	if(zip) {
		zip_source_t *src = zip_source_file(zip, file, 0, -1);
		if(zip == NULL) {
			show_error(awin, zip);
			zip_discard(zip);
			return FALSE;
		}

		err = zip_file_add(zip, FilePart(file), src, 0);
		if(err == -1) {
			show_error(awin, zip);
			zip_discard(zip);
			return FALSE;
		}

		err = zip_close(zip);
		if(err == -1) {
			show_error(awin, zip);
			zip_discard(zip);
			return FALSE;
		}
		return TRUE;
	}
```

show_error gets the error code and converts it to text; it is telling me "Renaming temporary file failed (file exists)"
I checked Snoopy and I can see a temporary file being renamed but no evidence that the original file has been deleted first.

**Comment by salass00** [comment]
The problem is probably that in posix the rename() function replaces the destination file if it already exists (unlike IDOS->Rename() which returns ERROR_OBJECT_EXISTS). The newlib rename() function only deletes the destination file first if unix semantics are enabled.

In filesysbox which is essentially an AmigaDOS to posix translation layer (as in it converts AmigaDOS file operations into posix ones) I have to explicitly check if the destination exists before calling rename() just so that it can fail with the correct error code.

**Comment by chris-y** [comment]
I guess this needs fixing in zip.library?  I could delete the original file in my code but that may not be safe to do.

**Comment by salass00** [comment]
Compiling your program with -lunix might also be an option, though probably not an appealing one if you are writing an AmigaOS native program.

**Comment by chris-y** [comment]
Yes, I'd rather not do that.

**Comment by salass00** [comment]
This issue should be fixed in the just released 54.1 version.

**Comment by chris-y** [comment]
Confirmed fixed, thanks.
